hazelcast/src/main/java/com/hazelcast/quorum/QuorumCallback.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.quorum;
+
+/**
+ * QuorumCallback can be used to notify quorum service for a particular quorum result that originated externally.
+ */
+public interface QuorumCallback {
```

Comment:
I find this a very hard to understand interface. Why is it called a call back? I make the implementation and then the callback is 'called back'? 

So is this an interface that you listen to changes? Or is this an interface where you control if a quorum is reached or not.